doc/scenario.md

```diff
@@ -32,6 +34,62 @@ start(Id) ->
     %% send some messages again
     ok.
 ```
+or, using the `start/2` function:
+
+```erlang
+-module(my_scenario).
+
+-export([init/0]).
+-export([start/2]).
+
+init() ->
+    %% initialize some metrics
+    Settings = get_settings(),
+    {ok, Settings}.
+
+start(Id, Settings) ->
+    %% connect user using Settings
+    %% fetch user's history
+    %% send some messages
+    %% wait a little bit
+    %% send some messages again
+    ok.
+```
+### Advanced features
+In addition to the required ones, `amoc_scenario` behaviour proposes optional callbacks:
+
+- `continue/0` and `terminate/0`
+
+    These callbacks make it possible to terminate a scenario at any moment.
+    The former checks if the current scenario is still valid, e.g. some metrics are
+    below a certain threshold.
+    It must return `continue` or `{stop, Reason}`.
+    The latter callback implements the actual termination.
+    It's called only if `continue/0` returned `{stop, Reason}`.
+    The reason is passed to the `terminate` callback.
+    When `continue` is returned nothing happens.
+    Only if both of them are implemented, automatic checking is enabled.
+    By default, it is scheduled to run every 60s.
+    This can be changed by setting the `scenario_checking_interval` application environment variable.
+    The checking logic happens in the `amoc_controller` process only on a
+    "master" Amoc node and only in the "distributed mode" (the scenario has
+    to be started via `amoc_dist:do/3/4`).
+
+-   `next_user_batch/2`
+    
+    This callback enables adding users in batches.
+    It is supposed to return strategy for adding users,
+    which is a list of `{Node, NumOfUsers, Interarrival}`.
+    It is then used by `amoc_controller:add_batches/2` function,
+    which takes the scenario module and the number of batches as parameters.
+    Parameters passed to the `next_user_batch` callback are 
```

Comment:
> The parameters passed [...]

(article)

It would be valid to omit the article only if we were talking about parameters in general, like, all parameters to all functions in the world (e.g. "Parameters are given to functions"), but we're talking about the specific parameters of this specific function, so, definite article 😛